### PR TITLE
feat(experiments): Closes 1084 Run MetadataService vs Embedly

### PR DESCRIPTION
**Pull request by sarracini**
- Sets up metadataService as an experiment
- Updates tests to mock the experiments bit

@jaredkerim r?

### CODE REVIEW

File: addon/PreviewProvider.js [hunk 1]
```diff
@@ -355,6 +356,11 @@ PreviewProvider.prototype = {
     this._metadataEndpoints.set("Embedly", simplePrefs.prefs["embedly.endpoint"]);
     this.enabled = simplePrefs.prefs[ENABLED_PREF];
     simplePrefs.on("", this._onPrefChange);
+
+    // if we are in the experiment change the metadata source
+    if (this._experimentProvider.data.metadataService) {
+      simplePrefs.prefs[METADATA_SOURCE_PREF] = "MetadataService";
```
Comment (jaredkerim) [inline addon/PreviewProvider.js]: Let's move all the hard coded strings of metadata source names to constants and refer to them that way.

File: test/test-PreviewProvider-MetadataStore.js [hunk 2]
```diff
@@ -119,11 +119,12 @@ function waitForAsyncReset() {
 }
 
 before(exports, function*() {
+  simplePrefs.prefs.metadataSource = "Embedly";
```
Comment (jaredkerim) [inline test/test-PreviewProvider-MetadataStore.js]: Should be a constant.

File: test/test-PreviewProvider.js [hunk 3]
```diff
@@ -423,29 +442,19 @@ exports.test_faulty_metadata_endpoint = function(assert) {
   assert.equal(gPreviewProvider._getMetadataSourceName(), gMetadataServiceSource, "properly set the source");
 };
 
+exports.test_metadata_service_experiment = function(assert) {
+  let mockExperimentProvider = {data: {metadataService: true}};
+  gPreviewProvider = new PreviewProvider(gMockTabTracker, gMockMetadataStore, mockExperimentProvider, {initFresh: true});
+  assert.equal(gPreviewProvider._getMetadataSourceName(), gMetadataServiceSource, "properly set the source if we are in the experiment");
```
Comment (jaredkerim) [inline test/test-PreviewProvider.js]: Preview Provider should set the pref, can we also check that the pref is set?

File: addon/PreviewProvider.js [hunk 4]
```diff
@@ -57,14 +58,14 @@ PreviewProvider.prototype = {
...
-    * pref, and falls back to Embedly if it doesn't exist
+    * pref or the experiment, and falls back to Embedly if it doesn't exist
     */
   _getMetadataSourceName() {
     let source = simplePrefs.prefs[METADATA_SOURCE_PREF];
     if (!this._metadataEndpoints.has(source)) {
       // set it to a default if the current endpoint was poorly set by the user
       // defensive programming ftw
-      return "Embedly";
+      source = "Embedly";
```
Comment (jaredkerim) [inline addon/PreviewProvider.js]: Should be a constant.

File: experiments.json [hunk 5]
```diff
@@ -28,5 +28,19 @@
...
+    "name": "Use Mozilla MetadataService",
+    "description": "Compute metadata using MetadataService",
+    "control": {
+      "value": false,
+      "description": "Use Embedly"
+    },
+    "variant": {
+      "id": "exp-003-metadata-service",
+      "value": true,
+      "threshold": 0.3,
```
Comment (sarracini) [inline experiments.json]: emtwo confirmed 0.3 is good

File: experiments.json [hunk 6]
```diff
@@ -28,5 +28,19 @@
       "threshold": 0.3,
       "description": "Show a recommendation in the third highlights spot"
     }
+  },
+  "metadataService": {
+    "name": "Use Mozilla MetadataService",
+    "description": "Compute metadata using MetadataService",
```
Comment (jaredkerim) [inline experiments.json]: Add active:false so we can land this disabled until the necessary server infrastructure is deployed and tested.

**Pull request merged**